curp/src/server/mod.rs

```diff
@@ -54,12 +54,21 @@ static DEFAULT_SERVER_PORT: u16 = 12345;
 
 /// The Rpc Server to handle rpc requests
 /// This Wrapper is introduced due to the `MadSim` rpc lib
-#[derive(Clone, Debug)]
+#[derive(Debug)]
 pub struct Rpc<C: Command, RC: RoleChange> {
     /// The inner server is wrapped in an Arc so that its state can be shared while cloning the rpc wrapper
     inner: Arc<CurpNode<C, RC>>,
 }
 
+impl<C: Command, RC: RoleChange> Clone for Rpc<C, RC> {
```

Comment:
Why not deriving from `Clone` directly?

Reply:
I got the following error when I try to add Clone to `#[derive]`:

```
error[E0599]: the method `clone` exists for struct `Rpc<C, RC>`, but its trait bounds were not satisfied
   --> curp/src/server/mod.rs:259:53
    |
58  | pub struct Rpc<C: Command, RC: RoleChange> {
    | ------------------------------------------
    | |
    | method `clone` not found for this struct
    | doesn't satisfy `Rpc<C, RC>: Clone`
...
259 |             .add_service(ProtocolServer::new(server.clone()))
    |                                                     ^^^^^ method cannot be called on `Rpc<C, RC>` due to unsatisfied trait bounds
    |
note: trait bound `RC: Clone` was not satisfied
   --> curp/src/server/mod.rs:57:17
    |
57  | #[derive(Debug, Clone)]
    |                 ^^^^^ unsatisfied trait bound introduced in this `derive` macro
help: consider restricting the type parameter to satisfy the trait bound
    |
58  | pub struct Rpc<C: Command, RC: RoleChange> where RC: Clone {
    |                                            +++++++++++++++

error[E0599]: the method `clone` exists for struct `Rpc<C, RC>`, but its trait bounds were not satisfied
   --> curp/src/server/mod.rs:305:53
    |
58  | pub struct Rpc<C: Command, RC: RoleChange> {
    | ------------------------------------------
    | |
    | method `clone` not found for this struct
    | doesn't satisfy `Rpc<C, RC>: Clone`
...
305 |             .add_service(ProtocolServer::new(server.clone()))
    |                                                     ^^^^^ method cannot be called on `Rpc<C, RC>` due to unsatisfied trait bounds
    |
note: trait bound `RC: Clone` was not satisfied
   --> curp/src/server/mod.rs:57:17
    |
57  | #[derive(Debug, Clone)]
    |                 ^^^^^ unsatisfied trait bound introduced in this `derive` macro
help: consider restricting the type parameter to satisfy the trait bound
    |
58  | pub struct Rpc<C: Command, RC: RoleChange> where RC: Clone {
    |                                            +++++++++++++++
```

It seems that rustc is not smart enough.